src/cpp/ext/otel/key_value_iterable.h

```diff
@@ -100,6 +103,22 @@ class OpenTelemetryPlugin::KeyValueIterable
         return false;
       }
     }
+    // Add per-call optional labels
+    if (record_optional_labels_) {
+      for (const auto& optional_label_key :
+           otel_plugin_->per_call_optional_label_keys_) {
+        auto it = optional_labels_.find(optional_label_key);
```

Comment:
Don't we need to check if the optional label is actually enabled here?  If it's enabled, then we need to add it here, regardless of whether the call tracer actually had a value for it; if it's not enabled, then we need to _not_ add it, even if the call tracer _does_ have a value for it.

Reply:
In the previous logic `per_call_optional_label_keys_` had the list of enabled optional labels, and we were iterating over those to make sure we were adding a value for all of them.

Reply:
In the previous logic, we only iterate over the set of enabled labels, so it works as expected. (Tested behavior.)

Logic is now updated to std::bitset.